### ISSUE TITLE
A few Bug/Enhancements?

### ISSUE THREAD

**Issue by BSarmady** [Kruithne/wow.export#426]
1- (Bug?) The exported bones doesn't seem right. There are some scattered bones around and their size doesn't look right either.

![image](https://github.com/user-attachments/assets/2198900d-61ee-4fd5-87c0-43b19824b53c)

2- (WIKI Enhancement?) Importing the model to Blender; Since I haven't seen bin type animation before, I'm not sure how I'm supposed to load them to blender. but the first animation that is loaded, seems to be missing keyframes even though animation plays (animation length is longer though)

3- (Bug) Separating the Character models looks nice at first, but when trying to select a character, things look like going haywire. Some parts does not have textures. some show texture without transparency (thought this was fixed about a year ago). and part names doesn't look right.  and some show incorrect texture (e.g. shows default skin color for parts like belt, pants, skirt, ...) and for some characters such as Dwarf Copter, shows incorrect part names, (or was it in the name list like that, which shows Facial for the balloon and wings?)

![image](https://github.com/user-attachments/assets/d400ecf5-da39-4fe5-966e-67940e19d885)

4- (Bug?) `Broken Race` female model seems to be broken as it only shows half of body. Is there another `broken race` model for temple NPCs than this one? I remember in one dungeon (or raid?) player would appear as broken, is this the model being used?

5- (Bug?) When exporting to glTF (apparently the only option at the moment) textures are missing. The only exported texture seems to be eyeglow and smoke (for dracthyr is 14 ) and name of textures are changed from what we previously had (from e.g. snow, base, ... to data01, data02 ...). This makes tracking textures harder.

![image](https://github.com/user-attachments/assets/fa17ccf0-b1ef-4184-b469-2c0fd0c0ac7b)

6- (Bug?) (I have Cata, Cata PTR, Classic, Classic PTR, DR, DR PTR 2 of them) installed in my system, in same folder and they share data folder (150GB) How can I select which one to be read by wow.export? Note that Cata added 4k Textures too but seems they are not showing in the texture list and I can't find classic human model in the characters either.

7- (Enhancement?) I couldn't find any option to show animations, is there a plan to add them, since you already can extract them and write them to files? (I couldn't figure out the animation spec from wow.dev otherwise I could help)

8- (bug?) App constantly trying to communicate with multicast (239.255.255.250) is it intentional?

**Comment by BSarmady** [comment]
> 1- that's related to the way Blender displays bones, not wow.export and not a bug.

I'm afraid it is not! Bones show the way they are defined, including size, direction and parent.
<img src="https://github.com/user-attachments/assets/6e379a9d-bab5-406d-84b3-f59aaee89b62"  width="300" />

> 2- Import the GLTF and the bin files are automatically dealt with, this is something you can google.

It seems your googling skills are better than mine. I couldn't find any reference on how `bin` is loaded with `glTF`. The only page I could find, was a Reddit post in blender group asking how to attach bin animation to model and someone high ranking guy asked him "what is bin?". Appreciate if you could provide link to the information you managed to find with searching google.

> 3- wow.export is for exporting, so only basic rendering is supported in the app. We do not have access to Blizzard names for geosets, so they are mostly based on characters.

I believe this specific rendering issue was reported [here](https://github.com/Kruithne/wow.export/issues/392) and fixed [here](https://github.com/Kruithne/wow.export/pull/413), but I'm guessing you already knew, since you made a comment in that issue.
<img src="https://github.com/user-attachments/assets/8ee3e72f-c23d-4656-8043-9570054b5288"  width="300" />. As for names, I guess its better call them `geosets` instead than calling them `ears`.

> 4- not a wow.export issue, that is how it is in the wow client from blizzard.

I'm assuming somehow it is shown as whole in the game client. If it is showing half then there should be an issue with reading the data which need to be investigated. some other problem that I can think of, is that the character model reference is wrong. If such character data does not exist, I assume, it should not show the `Type 2` in `body type` dropdown. BTW the character model for male valkyr  exist but female is missing.

> 5- ¯_(ツ)_/¯ I can't test this atm.

Well, Name of the png file is different from blp files used for texture (data0) for starter (in screenshot). Content of data*.png is merged texture from `bloodelffemaleskin00_*`, `bloodelffemale_hd_hair_style_*`, `bloodelffemale_hd_hair_color_*`, `bloodelffemale_hd_face_*`, `bloodelffemaleskin00_*`, `bloodelffemale_dh_blindfolds_*` and few others that I couldn't find. This makes re-texturing the model with existing textures a lot harder.
<img src="https://github.com/user-attachments/assets/97d354cd-5ca0-434c-ab0c-e2ecefa196b2"  width="200" />

> 6- Select the correct version after selecting the wow install.....

There are 3 issues. 
a) The button doesn't look like a button, it looks a lot like a grouping. so I wasn't expecting to be able to click it.
b) I clicked it and it shows a list, but names are not same as the ones shown in blizzard launcher (for example Cataclysm is missing or name is not correct)
<img src="https://github.com/user-attachments/assets/a7200e90-3ead-468a-8fd8-4434c6a1e5ab"  width="200" />
<img src="https://github.com/user-attachments/assets/61a7f4f1-a871-4787-a4b9-b4140391f361"  width="200" />
c) Trying to load Classic and Classic ptr, it gets stuck in loading listfile screen forever, I'm assuming there is no list file provided for those?
d) Apparently when selecting Wow latest, all textures from all games are shown, there are 4 different set of textures for bloodelffemale.

> 7- wow.export is for exporting, so supporting animation rendering is very low priority, if a priority at all.

Let me rephrase it. "List of animations just like the way characters are shown in a separate tabs" Some models have animations too and it seems they are not exported as those animations are shared between multiple models e.g. `7legionrifleman_m_AttackUnarmed_25`

> 8- ¯_(ツ)_/¯ provide more information.

I'm not sure what kind of more information you need. app is trying to connect to multicast twice a minute.
<img src="https://github.com/user-attachments/assets/faeb612f-eb5e-4860-9b14-07c13defd5b0"  width="200" />

**Comment by Marlamin** [comment]
1) Most exports aren't perfect and will require manual work to get to whatever standard you want them to be at. Stuff like bone rotations and all that is likely lost in translation between formats. This is something we could improve, but wow.export is in maintenance mode and not really under active development at this time.
2) glTF files use .bin as storage for binary data and are [part of the format](https://www.khronos.org/files/gltf20-reference-guide.pdf). Users never need to touch or open these, they are read by Blender's (or other) glTF importer. What it does with that is not something we control, but they should show up somewhere in Blender.
3) Most of the things you mentioned are either [known issues or feature requests](https://github.com/Kruithne/wow.export/issues/411) for wow.export's character support. For customizable mounts which are technically character models/races support is dodgy at best as we focused on actual player races on Retail (and it will be especially dodgy for ones added after we added character model support, e.g. the delve mount in the screenshot). 
4) This race and some other character races were implemented in WotLK and never finished by Blizzard but are listed in the character race database, so these models are broken in the client and as such in wow.export. We could filter these out manually, but if they were to magically fix them in the future we'd have to undo that, so that solution sucks.
5) This is working as intended, character models are complex and require pre-baking and blending of their textures before exporting. You can rebuild the textures manually from textures exported from the texture tab.
6) Optional AI upressed HD textures were added to Classic recently and are not something we plan to support.
a) This has not been a problem for other users.
b) We are not able to load the names the Blizzard launcher uses without a lot of effort or by manually keeping track, so we show the product names and versions. "World of Warcraft Classsic" is a perfectly valid name for Cata Classic, as it was WotLK Classic and TBC Classic before and will likely be MoP Classic in the not too distant future. We're not going to manually keep updating that each time. Users have so far been able to deduce which versions are which through the process of elimination/looking at the version numbers.
c) The listfile is shared between all WoW products but we save it per-build as it changes often, so it might take a while to download or might be blocked in certain regions and/or by firewalls/antivirus. If it remains stuck you a workaround you can do is downloading the latest listfile manually @ https://github.com/wowdev/wow-listfile/releases/latest/download/community-listfile.csv and changing the link to the listfile in wow.export settings to the location on disk where you saved community-listfile.csv. Keep in mind you will not receive future listfile updates with this method and newer files will be unnamed.
d) Character textures have many variants. Not sure what the point here is.
7) All available animations are exported, if the model does anything funky that we don't support then the animation might not work properly in Blender.
8) This is not something wow.export itself does but the Chromium/nwjs environment that it runs in does. This is not something we can control.

**Comment by Kruithne** [comment]
I am going to close this as it's a bit of a mega-issue. For any individual bugs/enhancements, can you please open individual issues for them if still applicable so we can track them properly. Thank you!
